accio-main/src/main/java/io/accio/main/wireprotocol/WireProtocolSession.java

```diff
@@ -333,14 +473,19 @@ public CompletableFuture<Optional<GenericTableRecordIterable>> sync()
     {
         CompletableFuture<Optional<GenericTableRecordIterable>> ended = runningQuery;
         runningQuery = CompletableFuture.completedFuture(null);
+        metadataQueries.clear();
         return ended;
     }
 
     public void close(byte type, String name)
+            throws Exception
     {
         switch (type) {
             case 'P':
-                portals.remove(name);
+                Optional.ofNullable(portals.get(name)).ifPresent(portal -> {
+                    portals.remove(name);
+                    metadataQueries.remove(preparedStmtPortalName(portal.getPreparedStatement().getName(), name));
+                });
```

Comment:
Curious, why not just use `Optional.ofNullable(portals.remove(name))`?

Reply:
I just forgot `remove` will return the portal. Thanks

Reply:
Let's make `remove` return nothing. It will be more clear.

Reply:
If PortalMap is like a Map, why we not make remove return value?

Reply:
To avoid to be ambiguous, and make the style be similar with `metadataQueries`.